File: day14/part2.py
input = bytearray(open('input').read().strip(),'UTF-8')

recipes = bytearray(b'37')
p1 = 0
p2 = 1

done = False
while (not done):
  n1 = recipes[p1]-ord('0')
  n2 = recipes[p2]-ord('0')
  for n in str(n1 + n2):
    recipes.append(ord(n))
    if (recipes[-len(input):] == input):
      done = True
      break
  p1 = (p1 + 1 + n1) % len(recipes)
  p2 = (p2 + 1 + n2) % len(recipes)

print(len(recipes) - len(input))

File: day14/part2.rb
input = IO.read('input').chomp

recipes = '37'
p1 = 0
p2 = 1

done = false
until done
  n1 = recipes[p1].to_i
  n2 = recipes[p2].to_i
  (n1 + n2).to_s.each_char do |n|
    recipes.concat(n)
    if recipes[-input.length..-1] == input
      done = true
      break
    end
  end
  p1 = (p1 + 1 + n1) % recipes.length
  p2 = (p2 + 1 + n2) % recipes.length
  # recipes.split('').each_index do |p|
  #  if p1 == p
  #    print '(' + recipes[p] + ')'
  #  elsif p2 == p
  #    print '[' + recipes[p] + ']'
  #  else
  #    print ' ' + recipes[p] + ' '
  #  end
  # end
  # puts
end

puts recipes.length - input.length
